homeassistant/components/airvisual/__init__.py

```diff
@@ -103,53 +116,77 @@ async def async_setup(hass, config):
     return True
 
 
-async def async_setup_entry(hass, config_entry):
-    """Set up AirVisual as config entry."""
+@callback
+def _standardize_geography_config_entry(hass, config_entry):
+    """Ensure that geography observables have appropriate properties."""
     entry_updates = {}
+
     if not config_entry.unique_id:
         # If the config entry doesn't already have a unique ID, set one:
         entry_updates["unique_id"] = config_entry.data[CONF_API_KEY]
     if not config_entry.options:
         # If the config entry doesn't already have any options set, set defaults:
-        entry_updates["options"] = DEFAULT_OPTIONS
+        entry_updates["options"] = {CONF_SHOW_ON_MAP: True}
+
+    if not entry_updates:
+        return
+
+    hass.config_entries.async_update_entry(config_entry, **entry_updates)
 
-    if entry_updates:
-        hass.config_entries.async_update_entry(config_entry, **entry_updates)
 
+async def async_setup_entry(hass, config_entry):
+    """Set up AirVisual as config entry."""
     websession = aiohttp_client.async_get_clientsession(hass)
 
-    hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id] = AirVisualData(
-        hass, Client(websession, api_key=config_entry.data[CONF_API_KEY]), config_entry
-    )
+    if CONF_API_KEY in config_entry.data:
+        _standardize_geography_config_entry(hass, config_entry)
+        airvisual = AirVisualGeographyData(
+            hass,
+            Client(websession, api_key=config_entry.data[CONF_API_KEY]),
+            config_entry,
+        )
+
+        # Only geography-based entries have options:
+        config_entry.add_update_listener(async_update_options)
+    else:
+        airvisual = AirVisualNodeProData(hass, Client(websession), config_entry)
 
     try:
-        await hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id].async_update()
-    except InvalidKeyError:
-        _LOGGER.error("Invalid API key provided")
+        await airvisual.async_update()
+    except NodeProError as err:
```

Comment:
When will this happen? It looks like we're catching these errors in `airvisual.async_update`.